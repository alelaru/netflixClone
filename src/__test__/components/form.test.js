import { render } from "@testing-library/react";
import Form from "../../components/form"
import { BrowserRouter as Router } from 'react-router-dom';

// const error = "NotFound"

jest.mock("react-router-dom")

describe("<Form/>", () => {

    it("Renders the <Form/> with populated data", () => {
        const {container, getByText, queryByText, getByPlaceholderText} = render(
                <Form>
                    <Form.Title>Sign In Now</Form.Title>
                    {/* {error && <Form.Error>{error}</Form.Error>} */}
                    {/* The on Submit we get rid of them */}
                    <Form.Base>
                        <Form.Input placeholder="Email Address" onChange={() => {}}></Form.Input>
                        <Form.Input placeholder="Password" type="password" onChange={() => {}}></Form.Input>
                        <Form.Submit disabled type="submit">Sign In</Form.Submit>
                    </Form.Base>
                    <Form.Text>
                        New to Netflix? <Router><Form.Link to={"/signup"}>Sign Up now.</Form.Link></Router>
                    </Form.Text> 
                    <Form.TextSmall>
                        This page is protected by Google Recaptcha to ensure you are not a bot
                    </Form.TextSmall>
                </Form>     
        ) 

        expect(queryByText("This page is protected by Google Recaptcha to ensure you are not a bot")).toBeTruthy();
        expect(getByText("Sign In Now")).toBeTruthy();
        expect(getByText("Sign In")).toBeTruthy();
        expect(getByText("Sign In").disabled).toBeTruthy();
        expect(getByPlaceholderText("Email Address")).toBeTruthy();
        expect(getByPlaceholderText("Password")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    })

    it("Renders the <Form/> with an error", () => {
        const {container, getByText, queryByText} = render(
                <Form>
                    <Form.Error>Your email address is already being used</Form.Error>
                    <Form.Submit type="submit">Sign In</Form.Submit>
                </Form>     
        ) 

        expect(queryByText("Your email address is already being used")).toBeTruthy();
        expect(getByText("Sign In").disabled).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    })

})