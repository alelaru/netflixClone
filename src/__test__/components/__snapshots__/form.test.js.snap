// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> Renders the <Form/> with an error 1`] = `
<div
  class="form__Container-sc-7952rh-0 SEPfC"
>
   
  <div
    class="form__Error-sc-7952rh-1 jmcCjM"
  >
    Your email address is already being used
  </div>
  <button
    class="form__Submit-sc-7952rh-8 kWcsU"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<Form/> Renders the <Form/> with populated data 1`] = `
<div
  class="form__Container-sc-7952rh-0 SEPfC"
>
   
  <h1
    class="form__Title-sc-7952rh-3 fCKYwV"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-7952rh-2 klyPny"
  >
    <input
      class="form__Input-sc-7952rh-7 gUryur"
      placeholder="Email Address"
    />
    <input
      class="form__Input-sc-7952rh-7 gUryur"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-7952rh-8 kWcsU"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-7952rh-4 fxUZEo"
  >
    New to Netflix? 
  </p>
  <p
    class="form__TextSmall-sc-7952rh-5 iEgvHc"
  >
    This page is protected by Google Recaptcha to ensure you are not a bot
  </p>
</div>
`;
